### PR TITLE
multiscale example

### PR DESCRIPTION
**Pull request by arnaudon**

### CODE REVIEW

**Review by d-schindler — COMMENTED** [review]
I think it would be good to define the seeds to make the example reproducible.

**Comment by arnaudon** [comment]
make plots of VI(ground_truth_i, best_comm_time) to show that at the scales we match each ground truth

**Comment by d-schindler** [comment]
I defined node_id's for the fine, middle and coarse scales which can be used for the _a posteriori_ comparison with the community id's from MS analysis.

File: examples/multiscale_example/multiscale_example.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    A[A < th] = 0.0
+    A[A > th] = 1.0
+    A = (A + A.T) / 2
+    return A
+
+
+if __name__ == "__main__":
+
+    # define root for storing figures
+    root = str(Path.cwd())+'/examples/multiscale_example/'
```
Comment (arnaudon) [inline examples/multiscale_example/multiscale_example.py:32]: remove this

File: src/pygenstability/pygenstability.py [hunk 2]
```diff
@@ -76,6 +77,7 @@ def run(
     result_file="results.pkl",
     n_workers=4,
     tqdm_disable=False,
+    with_optimal_scales = True,
```
Comment (arnaudon) [inline src/pygenstability/pygenstability.py:80]: add kwargs

File: src/pygenstability/pygenstability.py [hunk 3]
```diff
@@ -131,13 +133,17 @@ def run(
...
+            L.info("Compute ttprimes...")
             compute_ttprime(all_results, pool)
 
         if with_postprocessing:
             L.info("Apply postprocessing...")
             apply_postprocessing(all_results, pool, constructor=constructor)
 
+            if with_optimal_scales:
+                L.info("Identify optimal scales...")
+                all_results = identify_optimal_scales(all_results,window_size = int(0.1*n_time))
```
Comment (arnaudon) [inline src/pygenstability/pygenstability.py:145]: missing spaces for lint

File: examples/multiscale_example/multiscale_example.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    plt.figure()
+    plt.imshow(A)
+    plt.savefig(root + "adajcency_matrix.pdf", bbox_inches='tight')
+
+    # Multiscale structure
+    coarse_scale_id = np.zeros(n0)
+    middle_scale_id = np.zeros(n0)
+    fine_scale_id = np.zeros(n0)
+
+    for i in range(3):
```
Comment (arnaudon) [inline examples/multiscale_example/multiscale_example.py:78]: check this is correct

**Pull request merged**